learning/prompts/documentation-lookup/14_advanced_pipeline_patterns.md

```diff
@@ -0,0 +1,13 @@
+Prompt:
+What pipeline patterns exist in Apache Beam?
+Response:
+Beam pipeline patterns are a set of best practices for building Beam pipelines. They are based on real-world Beam deployments and are designed to help you build resilient, flexible, and portable Beam pipelines.
+
+Here are some of the most common pipeline patterns:
+- [File processing patterns](https://beam.apache.org/documentation/patterns/file-processing/)
+- [Pipeline options patterns](https://beam.apache.org/documentation/patterns/pipeline-options/)
+- [Side input patterns](https://beam.apache.org/documentation/patterns/side-inputs/)
+- [Windowing patterns](https://beam.apache.org/documentation/patterns/custom-windows/)
+- [State and Timers patterns](https://beam.apache.org/documentation/patterns/grouping-elements-for-efficient-external-service-calls/)
+
+For more information see the [Pipeline Patterns](https://beam.apache.org/documentation/patterns/overview/) section of the Beam programming guide.
```

Comment:
For more information, see the [Pipeline Patterns](https://beam.apache.org/documentation/patterns/overview/) section of the Beam programming guide.